sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -20,54 +19,108 @@
+access to features and gather community feedback.  Interfaces defined in this
+specification are implemented in {dpcpp}, but they are not finalized and may
+change incompatibly in future versions of {dpcpp} without prior notice.
+*Shipping software products should not rely on APIs defined in this
+specification.*
+
+== Backend support status
+This extension is currently implemented in {dpcpp} only for devices
+that contain a matrix hardware, specifically Intel(R) Advanced Matrix
+Extensions (Intel(R) AMX), Intel(R) Xe Matrix Extensions (Intel(R)
+XMX) and Nvidia(R) Tensor Cores.
+
+The joint_matrix type is an optional kernel feature as defined
+in section 5.7 of the core SYCL specification.  Each device supports
+only certain values for the `Rows` and `Cols` template parameters and
+only certain types for the `T` template parameter.  Applications can
+use the query API `tpu_params` to determine the set of legal
+parameters for each device.  If the application submits a kernel using
+an unsupported `joint_matrix` parameter, the implementation throws a
+synchronous exception with the `errc::kernel_not_supported` error code
+as described in section 5.7. 
+
+== Overview
+Joint matrix is a SYCL extension for matrix hardware programming. It
+unifies targets like Intel AMX in CPUs, Intel XMX in Intel GPUs and
+Nvidia Tensor Cores. This provides portable but performant API for
+users who want to build their own neural networks applications,
+perform custom optimzations, or experiment with new operations in a
+timely and performing manner.
+
+== Specification
+
+=== Feature test macro
 
-While this document presents the core API that unifies Intel AMX, Intel XMX, and Nvidia Tensor Cores, the implementations support slightly different versions of the API. For this reason, we introduce a new macro, namely `SYCL_EXT_ONEAPI_MATRIX_VERSION`  to distinguish between these different implementations. The goal in the next few months is to get rid of this implementation versioning macro. These are the current values for this macro.
+This extension provides a feature-test macro as described in the core SYCL
+specification. An implementation supporting this extension must predefine
+the macro `SYCL_EXT_ONEAPI_MATRIX` to one of the values defined in the
+table below. Applications can test for the existence of this macro to
+determine if the implementation supports this feature, or applications
+can test the macro's value to determine which of the extension's
+features the implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Matrix API Versions
+
+While this document presents the core API that unifies Intel AMX,
+Intel XMX, and Nvidia Tensor Cores, the implementations support
+slightly different versions of the API. For this reason, we introduce
```

Comment:
The situation is different now from what this paragraph states; because this document is specifically for the unified matrix interfaces which are portable. So I think it is best to replace this paragraph completely with e.g. the standard template for feature macro versioning.

Reply:
@JackAKirk, the standard template for feature macro versioning if not for experimental feature.
Once this moves out of experimental and becomes supported, this whole section "Matrix API versions" and SYCL_EXT_ONEAPI_MATRIX_VERSION macro will be removed. We won't need to keep the legacy API and tests. Right now, we only keep them to ensure current users have something working while we guide them through all these changes until we have something final (hopefully in this PR).

Do you suggest I remove this now?

Reply:
I just think that anyone reading this now as documentation on `joint_matrix` will be thrown by this

"the implementations support
slightly different versions of the API"

And since this does seem to be the main place that people will arrive at for `joint_matrix` documentation currently, it makes sense to address this now.

Reply:
I will remove it especially that we made value 4 as the default already.